codesign/README.md

```diff
@@ -1 +1,53 @@
-A standalone tool to codesign Mac engine binaries.
\ No newline at end of file
+# codesign
+
+A standalone tool to codesign Mac engine binaries.
+
+## Building
+
+This tool is meant to be published as an
+[AOT compiled binary](https://chrome-infra-packages.appspot.com/p/flutter/codesign)
+distributed via CIPD.
+
+Build the tool for different host platforms on corresponding machines. It will
+automatically download a suitable version of Dart to build the binary.
+
+To create the CIPD package, make sure that the `build/` folder does not exist.
+
+### Auto build
+
+Every new commit will trigger pre-submit builders to auto build a new version
+for different platforms without any tag/ref.
+
+When a new commit is submitted, post-submit builders will trigger a new version
+with a tag of `commit_sha`, and a ref of `staging`.
+
+### Manual build
+
+Running `tool/build.sh` will build an executable binary in
+the `build` folder. Then push to cipd by running
+
+```bash
+cipd create -in build                   \
```

Comment:
I uploaded a binary but it doesn't show up on https://chrome-infra-packages.appspot.com/p/flutter/codesign yet. Maybe it takes some extra time to propagate